x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/form/serializer.ts

```diff
@@ -16,8 +20,16 @@ const serializeAllocateAction = (
   originalActions: SerializedActionWithAllocation = {}
 ): SerializedActionWithAllocation => {
   const { allocate, migrate, ...rest } = newActions;
-  // First copy over all non-allocate and migrate actions.
-  const actions: SerializedActionWithAllocation = { allocate, migrate, ...rest };
+  // First copy over all non-require|include|exclude and migrate actions.
```

Comment:
What do you think of extracting this logic into a helper function, to encapsulate the operation? That would make it clearer to me what the operation is, and when reading the code I could examine it in isolation.

For example:

```ts
/**
 * Create an actions object with any actions that have been defined by ES but are unknown by the UI.
 */
const extractUnknownActions = (actions: SerializedActionWithAllocation) => {
  const { allocate, migrate, ...rest } = actions;
  // First copy over all non-allocate and migrate actions.
  const clonedActions: SerializedActionWithAllocation = { ...rest };

  // The UI only knows about include, exclude and require, so copy over all other values.
  if (allocate) {
    const { include, exclude, require, ...restAllocate } = allocate;
    if (!isEmpty(restAllocate)) {
      clonedActions.allocate = { ...restAllocate };
    }
  }

  return clonedActions;
};

/* snip */

// line 22
const actions = extractUnknownActions(newActions);
```

Note that I also reverted the "First copy over all non-require|include|exclude and migrate actions" comment because the phrasing describes logic a few lines down, but not this line, which I found confusing.